### DISCUSSION TITLE
rememberScreenModel issue

### DISCUSSION THREAD

**Discussion by alexhc23** [adrielcafe/voyager#331] [Q&A]
Hi :) hope you are doing well.

A few moments ago I just implemented navigation with Tabs. 
HomeScreen as the main container and other 4 tabs. 

I want to share data between screens but Tabs are objects so I can not share data through the constructor, so I tried with ScreenModel but it only works on the HomeScreen 

`
class HomeScreen(private val provider: DependencyProvider) : Screen {
    @OptIn(ExperimentalVoyagerApi::class)
    @Composable
    override fun Content() {

        val navigator = LocalNavigator.current
        val model = navigator?.rememberNavigatorScreenModel(
            tag = "HomeScreenModel",
            factory = { HomeScreenModel() }
        )
        model?.setHomeProvider(provider)

        val viewModel = kmpViewModel(
            factory = viewModelFactory { HomeViewModel(provider, model) }
        )

        TabNavigator(
            DashboardScreen,
            tabDisposable = {
                TabDisposable(
                    it,
                    listOf(DashboardScreen, OrdersScreen, ProfileScreen, SettingsScreen)
                )
            }
        ) {
            HomeScreenContent(it)

        }
    }
}
`

If I try to use the HomeScreenModel instance in other Tab, app crashes because I am not using the same instance

`
object OrdersScreen : Tab {

    override val options: TabOptions
        @Composable
        get() {
            val icon = rememberVectorPainter(Icons.Default.List)
            return remember {
                TabOptions(
                    index = 1u,
                    title = "Ordenes",
                    icon = icon
                )
            }
        }

    @OptIn(ExperimentalVoyagerApi::class)
    @Composable
    override fun Content() {
        val screenModel = LocalNavigator.currentOrThrow.rememberNavigatorScreenModel("HomeScreenModel"){
            HomeScreenModel()
        }
        val viewModel = kmpViewModel(
            factory = viewModelFactory {
                OrdersViewModel(
                    screenModel.getHomeProvider(),
                    screenModel.getUserType()
                )
            }
        )
        val state by viewModel.state.collectAsState()
        OrdersScreenContent(state, viewModel::onEvent)
    }
}
`

Not sure If it is a bug or I am not using the library properly.

Can you help me? please
Thanks in advance.

**Comment by Syer10** [comment]
The tabnavigator has its own navigator. So put all your HomeScreenModel things just before the HomeScreenContent
